docs/build-nix.md

```diff
@@ -0,0 +1,66 @@
+# Building on Nix
+
+The Nix Ecosystem insists on the purity of its builds, therefore compiling 
+Dosbox Staging on it requires slightly different inputs so meson gets the 
+appropriate environment variables from pkg-config.
+
+## Installing the Dependencies under Nix
+
+Nix has an unique toolset where you don't have to install it's packages to run 
+them. However, if you still want to, instead of using `nix-env` to install your 
+dependencies like your traditional package manager, we recommend you to do the 
```

Comment:
I'm having a hard time following this because the recommended approach is introduced without context as a compromised alternative:  _"You don't have to do A, but if you still want to, then instead of using B we recommend using C"_.

I would suggest only recommending one approach, and start with that:

Install dependencies using XYZ, as follows: `... commands ..`

If you want to learn more about the other ways to install NixOS depenencies, see: /link to all the alternatives/